www/js/lib/uiUtil.js

```diff
@@ -32,6 +32,48 @@ if (webpMachine) {
 }
 
 define(rqDef, function(settingsStore) {
+
+    /**
+     * Displays a Bootstrap alert or confirm dialog box depending on the options provided
+     * 
+     * @param {String} message The alert message to display in the body of the modal 
+     * @param {String} label The modal's label or title which appears in the header (optional, Default = "Confirmation" or "Message")
+     * @param {Boolean} isConfirm If true, the modal will be a confirm dialog box, otherwise it will be a simple alert message 
+     * @param {String} declineConfirmButton The text to display on the decline confirmation button (optional, Default = "Cancel") 
+     * @param {String} approveConfirmButton  The text to display on the approve confirmation button (optional, Default = "Confirm")
+     * @param {String} closeMessageButton  The text to display on the close alert message button (optional, Default = "Okay")
+     * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel/Okay, backdrop or the cross(x) button
+     */
+    function systemAlert(message, label, isConfirm, declineConfirmButton, approveConfirmButton, closeMessageButton) {
+        declineConfirmButton = declineConfirmButton || document.getElementById("declineConfirm").textContent;
+        approveConfirmButton = approveConfirmButton || document.getElementById("approveConfirm").textContent;
+        closeMessageButton = closeMessageButton || document.getElementById("closeMessage").textContent;
+        label = label || (isConfirm ? "Confirmation" : "Message");
+        return new Promise(function (resolve, reject) {
+            if (!message) reject("Missing body message");
+            // Set the text to the modal and it's buttons
+            document.getElementById("approveConfirm").textContent = approveConfirmButton;
```

Comment:
I think that this won't work in some cases.
If `systemAlert` is called twice in the same session:

- first time with a label override for one button (decline, for example). Let's say it passes the "No" label. This "No" label will be set in the DOM with `.textContent`
- second time without a text override. It will use the current content of the DOM, which is "No" instead of "Cancel"

What is missing is the code to revert the label to its original value after the modal is closed.

Again, if you don't feel comfortable with that, you can revert to the hardcoded version you had before. Never mind.